### PR TITLE
Migrating a php5app to php7 part three (Tools & Implementation)

### PR DESCRIPTION
**Pull request by unicodeveloper**

### CODE REVIEW

File: _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+## Work with First App
+
+We will build the first simple PHP 5 app very quickly. This is the scope of the app:
+
+* A user will be able to register on the app.
+* A user will be able to log into the app.
+* A user will be assigned a random Star Wars Code Name.
+* A user will be able to log out of the app.
+
+Building this app will require us to set up a database to store the users, write our registration and login code and manage the users session. Now, we won't employ the use of any framework because we don't want any form of overhead. Ordinarily, building this app will take a lot of time and setup but there is a service we can use to eliminate the hassle. Oh, yeah, Auth0 to the rescue!
```
Comment (kukicado) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: will take => would take

File: _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown [hunk 2]
```diff
@@ -0,0 +1,818 @@
...
+
+### Create and Configure Auth0 Client
+
+First thing we'll need to do is <a href="javascript:signup()">sign up for a free Auth0 account</a> and configure a new client.
+
+Now head over to [clients](https://manage.auth0.com/#/clients) and create a new one choosing 'Regular web Application' as the client type. Let's name it as something like 'Basic PHP WebApp'.
```
Comment (kukicado) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: clients => clients tab

File: _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown [hunk 3]
```diff
@@ -0,0 +1,818 @@
...
+```
+
+This checks if the user submitted a request to log out, clears the session and redirects the user back to the homepage.
+
+{% highlight html %}
+<script src="http://code.jquery.com/jquery-3.0.0.min.js" type="text/javascript"></script>
+<script src="https://cdn.auth0.com/js/lock/10.0/lock.min.js"></script>
+{% endhighlight %}
+
+We are making use of Auth0 Lock widget, and we also using jQuery to call the lock methods and handle button click event.
```
Comment (kukicado) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: Link to Lock landing page :)

File: _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown [hunk 4]
```diff
@@ -0,0 +1,818 @@
...
+
+{% highlight html %}
+<!-- font awesome from BootstrapCDN -->
+<link href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" rel="stylesheet">
+<link href="//maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css" rel="stylesheet">
+{% endhighlight %}
+
+
+Pulled in bootstrap and font-awesome for beautification.
```
Comment (kukicado) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: beautification :clap:

File: _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown [hunk 5]
```diff
@@ -0,0 +1,818 @@
...
+_Sign In_
+
+When you are logged in, you should be assigned a Star Wars codename like so:
+
+![Logged In](https://cdn.auth0.com/blog/app/loggedin.png)
+_Logged In_
+
+Our app is now running successfully on a PHP 5.x server. You can grab the [source code from Github](https://github.com/auth0-blog/starwars-phpapp) to ensure that everything works as expected.
+
+### Migrate to PHP 7
```
Comment (kukicado) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: Rename "Migrating our PHP5 App to PHP7"

File: _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown [hunk 6]
```diff
@@ -0,0 +1,818 @@
...
+
+We are currently running a PHP 5.x app. Let's migrate it to PHP 7. The good thing like I mentioned earlier in the previous post is that most times you might not have to change anything in the codebase. Let's see if that holds true for this app.
```
Comment (kukicado) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: Delete "like I mentioned earlier in the previous post"

File: _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown [hunk 7]
```diff
@@ -0,0 +1,818 @@
...
+![phpBB 3.1.3](https://cdn.auth0.com/blog/benchmark/phpBB.png)
+_phpBB 3.1.3_
+
+![Wordpress 4.1.1](https://cdn.auth0.com/blog/benchmark/wordpress.png)
+_Wordpress 4.1.1_
+
+![Drupal 8](https://cdn.auth0.com/blog/benchmark/drupal.png)
+_Drupal 8_
+
+From the results above, you can see that we can make double the amount of requests in lesser time in PHP 7 than PHP 5.
```
Comment (kukicado) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: lesser => less

File: _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown [hunk 8]
```diff
@@ -0,0 +1,818 @@
...
+Upgrade your server to at least PHP 7.0.0 and run this app again.
+
+![PHP 7 Server running](https://cdn.auth0.com/blog/app/php7status.png)
+_PHP 7 Server running_
+
+![Running on PHP 7](https://cdn.auth0.com/blog/app/php7_running_app.gif)
+_App running on PHP 7 without any errors_
+
+
+Awesome, now our first app is running on PHP 7 successfully!
```
Comment (kukicado) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: Suggestion: Is there anyway in the first app you can add a piece of code that would not work in PHP7. Doesn't have to be breaking change, could be warning, just to show the user how some of the tools from the beginning of the article can be used to easily fix type issues or other bugs from migrating.  What do you think?

Comment (unicodeveloper) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: Yeah, that absolutely makes sense!

File: _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown [hunk 9]
```diff
@@ -0,0 +1,818 @@
...
+
+![PHP 7 Readiness](https://cdn.auth0.com/blog/php/PHP7-Readiness-Inline-2.png)
+_Source: Jetbrains.com_
+
+The image below shows a typical example of an application that has classes with names that are reserved in PHP 7. Selecting **Run Inspection By Name** option from the **Code** menu, and then selecting the **PHP 7 Compatibility** section will give you results like this one below:
+
+![](https://cdn.auth0.com/blog/php/PHP7-Readiness-Results.png)
+_Source: Jetbrains.com_
+
+## Work with First App
```
Comment (unicodeveloper) [inline _posts/2017-01-26-migrating-a-php5-app-to-7-part-three.markdown]: Cool

**Pull request merged**